### PR TITLE
ui: add local search history and suggestions

### PR DESCRIPTION
**Pull request by harunurhan**
INSPIR-3408

### CODE REVIEW

File: ui/src/common/components/SearchBox/searchHistory.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { SEARCH_BOX_NAMESPACES } from '../../../reducers/search';
+import LRASet from './LRASet';
+
+const STORAGE_KEY = 'search-box-history';
+const HISTORY_LIMIT = 25;
+
+function runWhenIdle(task) {
```
Comment (drjova) [inline ui/src/common/components/SearchBox/searchHistory.js:7]: idle js everywhere

Comment (harunurhan) [reply ui/src/common/components/SearchBox/searchHistory.js:7]: this is when js engine is idle, not the user so unfortunately we can't use `idlejs` 😭  😃

**Comment by harunurhan** [comment]
Btw, I guess we don't care perfect multitab support, right? because currently it reads from storage only at the start and keeps write on it. 

If user was to search via multiple tabs, history would be overwritten always by the tab where the latest search happened and we wouldn't able to save searches from the other tab.

**Comment by drjova** [comment]
@harunurhan that's a good point, let's see if we get any compains about that

**Pull request merged**